src/types.ts

```diff
@@ -214,10 +208,7 @@ export type InferPageProps<
   Method extends keyof Controller,
 > = Controller[Method] extends (...args: any[]) => any
   ? Simplify<
-      Serialize<
-        InferProps<Extract<Awaited<ReturnType<Controller[Method]>>, PageObject>['props']> &
-          SharedProps
-      >
```

Comment:
The problem is that we need it here too. Otherwise SharedProps won't be available when using InferPageProps

Reply:
I see. Alright how about we keep the types as they are and just update the `config.stub`?
I'll update the PR so you can see an alternative aproach